tests/test_verbs/test_sample.py

```diff
@@ -0,0 +1,72 @@
+from clumper.clump import Clumper
+import pytest
+import itertools
+
+
+def test_oversampling(base_clumper):
+    with pytest.raises(ValueError):
+        base_clumper.sample(n=len(base_clumper) + 1, replace=False)
+
+
+def test_non_existent_weight(base_clumper):
+    with pytest.raises(ValueError):
+        base_clumper.sample(n=len(base_clumper) + 1, replace=False, weights="my-key")
```

Comment:
If you're testing for a mistake in the `weights` parameter, it would be good not to have a mistake in the `n` parameter. In your implementation the `n=len(base_clumper) + 1` parameter would also throw an error.

Reply:
That's a really good point!